scoring/score.py

```diff
@@ -2,15 +2,66 @@ class InvalidScoresheetException(Exception):
     pass
 
 
+CAN_STATE_SCORES = {
+    "B": 3,  # 3 points for can with tape on the Bottom
+    "T": 1,  # 1 point for can with tape on the Top
+    "S": 0,  # 0 points for can on its Side
+}
+
+MAX_CANS = 28
+
+
 class Scorer(object):
     def __init__(self, teams_data, arena_data):
         self._teams_data = teams_data
+        self._arena_data = arena_data
 
     def calculate_scores(self):
-        raise NotImplementedError
+        scores = {}
+        for tla, info in self._teams_data.items():
+            zone = info['zone']
+            cans = self._arena_data[zone]['tokens']
+            points = sum(CAN_STATE_SCORES[c] for c in cans if c != " ")
+
+            if info.get('left_scoring_zone', False):
+                points += 1
+
+            scores[tla] = points
+
+        return scores
 
     def validate(self, extra_data):
-        raise NotImplementedError
+        cans = "".join(
+            info['tokens']
+            for info in self._arena_data.values()
+        )
```

Comment:
YAML being YAML I think we're likely to have non-string values turn up here, maybe some separate validation or `str()`?

Reply:
We could validate it for sure, though I think it's fine for such things here to just error.